test-js/unit/pentaho/util/promise.Spec.js

```diff
@@ -0,0 +1,316 @@
+              expect(reason).toBe(error);
+              done();
+            });
+      });
+
+      it("should call fun on the given ctx object", function() {
+        var fun = jasmine.createSpy("promiseWrapCall");
+        var ctx = {};
+        promiseUtil.wrapCall(fun, ctx);
+        expect(fun.calls.count()).toBe(1);
+        expect(fun.calls.first().object).toBe(ctx);
+      });
+    });
+
+    describe("require(deps, localRequire)", function() {
+      it("should throw when `deps` is not specified", function() {
+        function expectIt(args) {
+          expect(function() {
+
+            promiseUtil.require.apply(promiseUtil, args);
+
+          }).toThrowError(error.argRequired("deps").message);
+        }
+
+        expectIt([]);
+        expectIt([null]);
+        expectIt([undefined]);
+      });
+
+      it("should fulfill to a single value when `deps` is a string", function(done) {
+        var mid = "pentaho/tests/promise/require/test1";
+        require.undef(mid);
+        var moduleValue = {};
+        define(mid, [], function() {
+          return moduleValue;
+        });
+
+        var promise = promiseUtil.require(mid);
+        expect(promise instanceof Promise).toBe(true);
+        promise.then(function(value) {
+          expect(value).toEqual(moduleValue);
+          require.undef(mid);
+          done();
+        }, done.fail);
+      });
+
+      it("should fulfill to an empty array when `deps` is an empty array", function(done) {
+        var promise = promiseUtil.require([]);
+        expect(promise instanceof Promise).toBe(true);
+        promise.then(function(values) {
+          expect(values).toEqual([]);
+          done();
+        }, done.fail);
+      });
+
+      it("should fulfill to an array value when `deps` is a (non-empty) array", function(done) {
+        var mid1 = "pentaho/tests/promise/require/test1";
+        var mid2 = "pentaho/tests/promise/require/test2";
+        require.undef(mid1);
+        require.undef(mid2);
+
+        var moduleValue1 = {};
+        var moduleValue2 = {};
+        define(mid1, [], function() {
+          return moduleValue1;
+        });
+        define(mid2, [], function() {
+          return moduleValue2;
+        });
+
+        var promise = promiseUtil.require([mid1, mid2]);
+        expect(promise instanceof Promise).toBe(true);
+        promise.then(function(values) {
+          expect(Array.isArray(values)).toBe(true);
+          expect(values.length).toBe(2);
+          expect(values[0]).toBe(moduleValue1);
+          expect(values[1]).toBe(moduleValue2);
+          require.undef(mid1);
+          require.undef(mid2);
+          done();
+        }, done.fail);
+      });
+
+      it("should use the specified `localRequire` instead of the global one when `deps` is a string", function() {
+        // Executed synchronously
+        var localRequire = jasmine.createSpy("localRequire")
+            .and.callThrough(function(deps, calllback, errback) {
```

Comment:
The deliberate misspelled argument `calllback` can be easily missed, and might lead to unnecessary wasting of time.
What about renaming the argument to `f`?